checker/src/org/checkerframework/checker/index/upperbound/UpperBoundAnnotatedTypeFactory.java

```diff
@@ -460,10 +467,15 @@ public Void visitBinary(BinaryTree tree, AnnotatedTypeMirror type) {
             ExpressionTree right = tree.getRightOperand();
             switch (tree.getKind()) {
                 case PLUS:
-                case MINUS:
                     // Dataflow refines this type if possible
                     type.addAnnotation(UNKNOWN);
                     break;
+                case MINUS:
+                    // Dataflow refines this type most of the time.
+                    // But when a NonNegative is subtracted from a LTL,
+                    // the result is still LTL without an offset.
+                    addAnnotationForMinus(left, right, type);
```

Comment:
This needs to be done in dataflow or else information about offsets will be lost.  See test cases I added: https://github.com/typetools/checker-framework/pull/1068/commits/aaa2cabd04d9e47a0b867e369cb9cb652d28dfca